app/bundles/UserBundle/Security/Authenticator/PluginAuthenticator.php

```diff
@@ -93,7 +98,11 @@ public function authenticate(Request $request): Passport
             }
         }
 
-        if (!$user instanceof User && !$authenticated && null === $response) {
+        if (!$authenticated && $response) {
+            throw new RespondingAuthenticationException($response);
```

Comment:
Wasn't sure how else to return this as it looks like can't really return a passport as there's no user known until it returns. In v5 it would return a plugin token with no user and then use the listener to use the response. This method allows the `onAuthenticationFailure` to process it.